PelicanPlatform/pelican
# Director-based origin health tests

Closes #175 

## Test Instructions

Make sure in your `pelican.yaml`, the host of your `Federation.DirectorUrl` is the same as your `hostname`. If you run things in container, it should be a string instead of `localhost`

Have director & registry & origin running at debug mode (`-d`) and watch for the log message at director's end: 

```console
DEBUG[2023-11-07T14:48:11Z] Starting Director test for origin fae8c2865de4 at https://fae8c2865de4:8443 
DEBUG[2023-11-07T14:48:41Z] Starting a new Director test cycle for origin: fae8c2865de4 at https://fae8c2865de4:8443 
......
DEBUG[2023-11-07T14:48:41Z] Director test cycle succeeded at Tue Nov  7 14:48:41 UTC 2023
```

Also, at origin's end, the console message should reflect that the director uploaded/downloaded/deleted a file to/from origin and also you **shouldn't** see any errors from Xrootd suggesting a file upload is failed:

```console
INFO[2023-11-07T14:48:41Z] Served Request                                client=172.17.0.2 daemon=gin fields.time=1.792959ms method=POST resource=/api/v1.0/origin-api/directorTest status=200
```
